hash/hash.go

```diff
@@ -93,6 +93,15 @@ func Decode(h string) (Hash, error) {
 	return Hash(hash), nil
 }
 
+// DecodeFromBytes decodes hash and checks it length.
+// It returns empty hash on nil slice of bytes.
+func DecodeFromBytes(data []byte) (Hash, error) {
```

Comment:
This function should be called by a bunch of function of this file that use the pattern:
```go
	if len(data) != size {
		return nil, fmt.Errorf("hash: invalid length")
	}
	return Hash(data), nil
```
functions: `Decode`, `Unmarshal`, `UnmarshalJSON` should also

Reply:
There is an update in Execution SDK that also modifies that. I will refactor it after Execution and this PR will merge

Reply:
It's merged since 3 days ;)
https://github.com/mesg-foundation/engine/pulls?q=is%3Apr+is%3Aclosed+sort%3Aupdated-desc

Reply:
I looked at it and I need to rethink the api. Because some of them (like Marshal) requires make and copy slice,and some don't